pkg/auth/user/required_attribute_verifier.go

```diff
@@ -0,0 +1,39 @@
+package user
+
+import (
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/sliceutils"
+)
+
+// NewRequiredAttributesVerifier returns an AttributeVerifier that will verify all
+// attributes of a user and return an error if any of the required attributes are missing or
+// have a value that is different than expected.
+func NewRequiredAttributesVerifier(requiredAttributes []*storage.AuthProvider_RequiredAttribute) AttributeVerifier {
+	return &checkRequiredAttributesImpl{
+		required: requiredAttributes,
+	}
+}
+
+type checkRequiredAttributesImpl struct {
+	required []*storage.AuthProvider_RequiredAttribute
+}
+
+func (c checkRequiredAttributesImpl) Verify(attributes map[string][]string) error {
```

Comment:
can we make it pointer receiver function?

Reply:
Why? 
It's not required to be used here as we never make changes (and also shouldn't) to the list of required attributes.

Reply:
My understanding is that value receiver would result in an additional copy of `checkRequiredAttributesImpl` instance on every `Verify` call. Could safe some bits

Reply:
Feel like with the small set of array elements we deal with it wouldn't necessarily safe anything, but the mantra is anyways when in doubt use pointer receivers  🤷  -- so lets go ahead and use them!